test/integration/physics_system.cc

```diff
@@ -875,3 +875,110 @@ TEST_F(PhysicsSystemFixture, GetBoundingBox)
       ignition::math::Vector3d(-0.25, 2, 1)),
       bbox.begin()->second);
 }
+
+
+/////////////////////////////////////////////////
+// This tests whether nested models can be loaded correctly
+TEST_F(PhysicsSystemFixture, NestedModel)
+{
+  ignition::gazebo::ServerConfig serverConfig;
+
+  const auto sdfFile = std::string(PROJECT_SOURCE_PATH) +
+    "/test/worlds/nested_model.sdf";
+
+  sdf::Root root;
+  root.Load(sdfFile);
+  const sdf::World *world = root.WorldByIndex(0);
+  ASSERT_TRUE(nullptr != world);
+
+  serverConfig.SetSdfFile(sdfFile);
+
+  gazebo::Server server(serverConfig);
+
+  server.SetUpdatePeriod(1us);
+
+  // Create a system that records the poses of the links after physics
+  test::Relay testSystem;
+
+  std::unordered_map<std::string, ignition::math::Pose3d> postUpModelPoses;
+  std::unordered_map<std::string, ignition::math::Pose3d> postUpLinkPoses;
+  std::unordered_map<std::string, std::string> parents;
+  testSystem.OnPostUpdate(
+    [&postUpModelPoses, &postUpLinkPoses, &parents](const gazebo::UpdateInfo &,
+    const gazebo::EntityComponentManager &_ecm)
+    {
+      _ecm.Each<components::Model, components::Name, components::Pose>(
+        [&](const ignition::gazebo::Entity &_entity, const components::Model *,
+        const components::Name *_name, const components::Pose *_pose)->bool
+        {
+          // store model pose
+          postUpModelPoses[_name->Data()] = _pose->Data();
+
+          // store parent model name, if any
+          auto parentId = _ecm.Component<components::ParentEntity>(_entity);
+          if (parentId)
+          {
+            auto parentName =
+                _ecm.Component<components::Name>(parentId->Data());
+            parents[_name->Data()] = parentName->Data();
+          }
+          return true;
+        });
+
+      _ecm.Each<components::Link, components::Name, components::Pose,
+                components::ParentEntity>(
+        [&](const ignition::gazebo::Entity &, const components::Link *,
+        const components::Name *_name, const components::Pose *_pose,
+        const components::ParentEntity *_parent)->bool
+        {
+          // store link pose
+          postUpLinkPoses[_name->Data()] = _pose->Data();
+
+          // store parent model name
+          auto parentName = _ecm.Component<components::Name>(_parent->Data());
+          parents[_name->Data()] = parentName->Data();
+          return true;
+        });
+
+      return true;
+    });
+
+  server.AddSystem(testSystem.systemPtr);
+  server.Run(true, 1, false);
+
+  EXPECT_EQ(2u, postUpModelPoses.size());
+  EXPECT_EQ(2u, postUpLinkPoses.size());
+  EXPECT_EQ(4u, parents.size());
```

Comment:
found some test failures here 
```
/home/claire/citadel_ws/src/ign-gazebo/test/integration/physics_system.cc:949: Failure
Value of: postUpModelPoses.size()
  Actual: 1
Expected: 2u
Which is: 2
/home/claire/citadel_ws/src/ign-gazebo/test/integration/physics_system.cc:950: Failure
Value of: postUpLinkPoses.size()
  Actual: 1
Expected: 2u
Which is: 2
/home/claire/citadel_ws/src/ign-gazebo/test/integration/physics_system.cc:951: Failure
Value of: parents.size()
  Actual: 2
Expected: 4u
Which is: 4
/home/claire/citadel_ws/src/ign-gazebo/test/integration/physics_system.cc:958: Failure
Expected: (postUpModelPoses.end()) != (modelIt), actual: 8-byte object <00-00 00-00 00-00 00-00> vs 8-byte object <00-00 00-00 00-00 00-00>

```

Reply:
hmm weird I'm not able to reproduce this. Will wait for another jenkins results and see if the builds catch this as well.

Reply:
I updated all repo and reran the tests and the failure went away!